src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/ElasticSearchConfiguration.java

```diff
@@ -132,8 +132,10 @@ public Integer getConnectionTimeoutMillis() {
         return connectionTimeoutMillis;
     }
 
-    public Integer getConnectionTimeoutMillisOrDefault() {
-        return (connectionTimeoutMillis == null || connectionTimeoutMillis < -1) ? CONNECTION_TIMEOUT_DEFAULT : connectionTimeoutMillis;
+    public int getConnectionTimeoutMillisOrDefault() {
+        if(connectionTimeoutMillis == null) return CONNECTION_TIMEOUT_DEFAULT;
+        if(connectionTimeoutMillis < -1) return -1;
```

Comment:
is -1 a valid value?
0 means to wait indefinitely for normal Sockets

Reply:
The API we call in the end allows -1 and iterprets it as "undefined". But I removed this option now.